src/test/scala/scorex/core/DefaultModifiersCacheSpecification.scala

```diff
@@ -0,0 +1,120 @@
+package scorex.core
+
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatest.prop.{GeneratorDrivenPropertyChecks, PropertyChecks}
+import scorex.core.consensus.{History, HistoryReader, ModifierSemanticValidity, SyncInfo}
+import scorex.core.consensus.History.ModifierIds
+import scorex.core.serialization.Serializer
+import scorex.crypto.hash.Blake2b256
+
+import scala.util.Try
+
+class DefaultModifiersCacheSpecification extends PropSpec
+  with PropertyChecks
+  with GeneratorDrivenPropertyChecks
+  with Matchers {
+
+  property("ignore adding self as a peer") {
+    class FakeModifier extends PersistentNodeViewModifier {
+      override def parentId: ModifierId = ???
+
+      override val modifierTypeId: ModifierTypeId = ModifierTypeId @@ (0: Byte)
+
+      override def id: ModifierId = ???
+
+      override type M = this.type
+
+      override def serializer: Serializer[FakeModifier.this.type] = ???
+    }
+
+    class FakeSyncInfo extends SyncInfo {
+      override def startingPoints: ModifierIds = ???
+
+      override type M = this.type
+
+      override def serializer: Serializer[FakeSyncInfo.this.type] = ???
+    }
+
+    class FakeHr extends HistoryReader[FakeModifier, FakeSyncInfo] {
+      /**
```

Comment:
I think you can remove comments from non-implemented methods in "fake" classes.  They are not going to be used by the tests anyway.

Reply:
+1. And may be it's better to extract this helper classes to directory with mocks